src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -186,280 +214,319 @@ def slice_info_for_literal(self, length, prefix="") -> RawSliceInfo:
+                unique_alternate_id, alternate_code
+            )
+
+    def update_next_slice_indices(
+        self, block_idx: int, block_type: str, tag_name: str
+    ) -> None:
+        """Based on block, update conditional jump info."""
+        if block_type == "block_start" and tag_name in (
+            "for",
+            "if",
+        ):
+            self.stack.append(block_idx)
+        elif block_type == "block_mid":
+            # Record potential forward jump over this block.
+            self.raw_slice_info[
+                self.raw_sliced[self.stack[-1]]
+            ].next_slice_indices.append(block_idx)
+            self.stack.pop()
+            self.stack.append(block_idx)
+        elif block_type == "block_end" and tag_name in (
+            "endfor",
+            "endif",
+        ):
+            if not self.inside_set_or_macro:
+                # Record potential forward jump over this block.
+                self.raw_slice_info[
+                    self.raw_sliced[self.stack[-1]]
+                ].next_slice_indices.append(block_idx)
+                if self.raw_sliced[self.stack[-1]].slice_subtype == "loop":
+                    # Record potential backward jump to the loop beginning.
+                    self.raw_slice_info[
+                        self.raw_sliced[block_idx]
+                    ].next_slice_indices.append(self.stack[-1] + 1)
+                self.stack.pop()
+
+    def handle_left_whitespace_stripping(self, token: str) -> None:
+        """If block open uses whitespace stripping, record it.
+
+        When a "begin" tag (whether block, comment, or data) uses whitespace
+        stripping
+        (https://jinja.palletsprojects.com/en/3.0.x/templates/#whitespace-control)
+        the Jinja lex() function handles this by discarding adjacent whitespace
+        from 'raw_str'. For more insight, see the tokeniter() function in this file:
+        https://github.com/pallets/jinja/blob/main/src/jinja2/lexer.py
+
+        We want to detect and correct for this in order to:
+        - Correctly update "idx" (if this is wrong, that's a potential
+          DISASTER because lint fixes use this info to update the source file,
+          and incorrect values often result in CORRUPTING the user's file so
+          it's no longer valid SQL. :-O
+        - Guarantee that the slices we return fully "cover" the contents of
+          'in_str'.
+
+        We detect skipped characters by looking ahead in in_str for the token
+        just returned from lex(). The token text will either be at the current
+        'idx_raw' position (if whitespace stripping did not occur) OR it'll be
+        farther along in 'raw_str', but we're GUARANTEED that lex() only skips
+        over WHITESPACE; nothing else.
+        """
+        # Find the token returned. Did lex() skip over any characters?
+        num_chars_skipped = self.raw_str.index(token, self.idx_raw) - self.idx_raw
+        if num_chars_skipped:
```

Comment:
Personally Id prefer a guard `if not num_chars_skipped: \ return`

Reply:
I'll do that. Previously, I had just mechanically pasted this code after pulling it out of a larger function. 👍